### ISSUE TITLE
DependsOn ordering in tests

### ISSUE THREAD

**Issue by andrew-glenn** [aws-quickstart/taskcat#419]
Migrating [this issue from the taskcat/taskcat-v9 repo,](https://github.com/taskcat/taskcat/issues/32) as it's now archived.

> A need has been identified to serialze test runs via DependsOn. The use-case involves using multiple CFN Stacks to 1) build infra, and 2) run integration tests. Running the integration tests in parallel as the infra is being built does not lead to a desired outcome.

from @jaymccon :

> thinking about this, implicit dependencies should be pretty easy to do if we have a new magic parameter, something like $[taskcat_output_MyTestName_MyOutputName] would fetch the MyOutputName output from the stack created by the MyTestName test

**Comment by jeffreycoe** [comment]
Hi @andrew-glenn - Do you happen to know when Taskcat 1.0 might be released with this feature?

**Comment by andrew-glenn** [comment]
@jeffreycoe the original intent was that v1.0 would be in 2020. However, with everything else going on in the world these days, that timeline isn't something that we feel we can meet. 

However - if you have a need for it, especially if it would empower production usage, I can prioritize specific features such as this ahead of v1.0. I don't have an ETA either way but context will enable me to shuffle things around.

**Comment by jeffreycoe** [comment]
@andrew-glenn We’re building out our CI pipelines now for our CloudFormation templates, and have been investigating using Taskcat to help drive our CI tests.  One specific use case we have is around EKS.  We were attempting to use Taskcat to test our EKS worker node stacks, but it’s dependent on having an EKS cluster deployed first. Since we don’t have this feature, we’re stuck with creating custom scripts to run our tests.

I completely understand how priorities can change, though. This would definitely assist us with our pipelines by simplifying them, and would provide some help with getting to production. As we’re just one AWS user out of the many, I would not want you to reprioritize other higher priority work for this one feature if it’s not a critical feature the rest of the community needs right now. Also, I can try to assist with this in any way I can (although my python skills are a bit rusty right now).

**Comment by andrew-glenn** [comment]
@jeffreycoe I'm just now seeing this, but it helps identify the pain you're experiencing without it.

In your example, your tests would first deploy an EKS cluster with a test named `eks-cluster-provisioning`, then when that passed, provision `eks-worker-node-stacks` using `DependsOn`?.

**Comment by jeffreycoe** [comment]
@andrew-glenn Yes, exactly. Also as an example, the AWS Quick Start templates use CloudFormation custom resources to do some configuration with the EKS cluster once it's online. We'd have to use `DependsOn` on those custom resources as we'd have to wait for the EKS cluster resource to complete its provisioning process before they can be executed.

Here are some example templates: https://github.com/aws-quickstart/quickstart-amazon-eks/tree/main/templates

**Comment by t0ffel** [comment]
We use cloudformation to do layered deployments. i.e. separate cfn templates are used to create separate layers.
`DependsOn` along with being able to take output and use it in the dependent test would work great for us.
